### DISCUSSION TITLE
macOS: Searching box loses focus after entering a single character on tab overview

### DISCUSSION THREAD

**Discussion by jaycem-dev** [ghostty-org/ghostty#3556] [Feature Requests, Ideas] [answered]
### How to reproduce
- Enter tab overview.
- Start typing the name of the tab, the search box loses focus after the first character unless manually selected.

I also can't find a way to select a tab to enter after searching, but this could be a different issue.

**Comment by czuniga9** [comment]
I noticed this too, typing anything starts the filter but the box loses focus so you are only able to type a single character.

You can highlight a tab with `ctrl+tab` (forward), `ctrl+shift+tab` (backward) and then `esc` for selecting it. I'd really like to be able to change the keybindings for this so I can use arrow keys and enter instead (should I open a new discussion/issue for this?).

**Comment by liby** [comment]
Is the `toggle_tab_overview` action available on macOS? I set the 

```
keybind = super+i=toggle_tab_overview
```
and found it didn't work. After looking at the code, I also found that this feature is [not implemented on macOS](https://github.com/ghostty-org/ghostty/blob/main/macos/Sources/Ghostty/Ghostty.App.swift#L546-L555).

**Reply by jaycem-dev** [reply]
@liby The functionality is available on the 1.0 release at least, I don't know if they removed it from the docs...
I'm using the default keybind but you can select the option from the menubar to test it (View -> Show All Tabs)

![image](https://github.com/user-attachments/assets/210b43be-e950-47a4-85ac-b83d7a5eca74)
![image](https://github.com/user-attachments/assets/5577e036-c4fe-4815-bd8b-4057aac68726)

You also need to have window decorations enabled.

**Reply by liby** [reply]
I think this might actually be [macOS's native "Show All Tabs" feature](https://support.apple.com/guide/mac-help/use-tabs-in-windows-mchla4695cce/mac#:~:text=Show%20and%20navigate%20tabs) rather than Ghostty's `toggle_tab_overview` action. 

This would explain why it works with the system shortcut (`Shift + Cmd + \`) and through the native menu bar, and yes, as you mentioned, window decorations need to be enabled since this relies on macOS's native window management system.

**Comment by goproslowyo** [comment]


**Reply by liby** [reply]
https://github.com/ghostty-org/ghostty/discussions/3556#discussioncomment-11701917

**Reply by goproslowyo** [reply]
Ah yeap, `toggle_tab_overview` is the Linux/GTK whereas macOS does this natively with `cmd+shift+\` (try other windows like finder, too).

**Comment by 00-kat** [comment, marked as answer]
As @liby explained in https://github.com/ghostty-org/ghostty/discussions/3556#discussioncomment-11701917, that tab overview is built-in to macOS.  Does this issue appear if you toggle the tab overview in another app which has native tabs, like, say, Finder?  If so, it's likely an Apple bug.

**Reply by bdcarr** [reply]
It does 😭 what a shame. If it was a ghostty bug I'm sure it'd have been fixed by now

**Reply by 00-kat** [reply]
I'll close this discussion in that case, since we sadly can't fix it.
